src/dev/build/tasks/nodejs_modules/clean_client_modules_on_dll_task.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { getDllEntries, cleanDllModuleFromEntryPath, writeEmptyFileForDllEntry } from './webpack_dll';
+import { getDependencies } from './get_dependencies';
+import globby from 'globby';
+
+export const CleanClientModulesOnDLLTask = {
+  description:
+    'Cleaning client node_modules bundled into the DLL',
+
+  async run(config, log, build) {
+    const baseDir = build.resolvePath('.');
+    const kbnPkg = config.getKibanaPkg();
+    const kbnPkgDependencies = (kbnPkg && kbnPkg.dependencies) || {};
+    const kbnWebpackLoaders = Object.keys(kbnPkgDependencies).filter(dep => !!dep.includes('-loader'));
+    const mainCodeEntries = [
+      build.resolvePath('src/cli'),
+      build.resolvePath('src/cli_keystore'),
```

Comment:
I am concerned that the definition here for entry files is not an obvious one when someone would want to add a new entry and not have the dependencies stripped when building. 

Thoughts on centralizing these values to a definition somewhere? In addition, having some documentation outlining how the DLL process works would be helpful.

Reply:
@tylersmalley I've added documentation into `src/dev/build/README.md`

Reply:
@tylersmalley I have also applied some simplification for the code building the server code entries. If we found this problematic in the future, I think it will be more easy to load those paths from a config file. However, I don't think this will be a problem, as those changes won't happen too much as they are structural changes.